nano/lib/blocks.cpp

```diff
@@ -357,6 +357,11 @@ bool nano::send_block::valid_predecessor (nano::block const & block_a) const
 	return result;
 }
 
+nano::epoch nano::send_block::epoch () const
```

Comment:
Because it's similar for send/open/change/receive types, then probably it can be just common `nano::epoch nano::block::epoch () const` with override for state_block (like nano::block::link (), account (), representative ())